### ISSUE TITLE
Include "Approved" flag in Pull Requests

### ISSUE THREAD

**Issue by kminehart** [grafana/github-datasource#70]

**Comment by vschettino** [comment]
Hey @kminehart I don't think there is a "Approved" flag in the [Pull Request object](https://docs.github.com/en/graphql/reference/objects#pullrequest). Maybe we could add a similar feature using the [review](https://docs.github.com/en/graphql/reference/objects#pullrequestreview) state?

**Comment by maximbaz** [comment]
That seems to be what Github staff [is suggesting](https://github.community/t/graphql-v4-pullrequest-approved/14045/2):

> A pull request is approved when there is one or more reviews with an `APPROVED` state. So you would use the `reviews` connection, filter by `APPROVED`, and check if the `totalCount` is greater than 0.

**Comment by kminehart** [comment]
interesting. That shouldn't be too hard. But probably not easy enough to be a "good first issue" :P

I'll take a look this weekend.

**Comment by vschettino** [comment]
I don't know guys, there are ways to configure how many approvals are required for merge, or if new commits needs a new approval for merge:
- https://docs.github.com/en/github/administering-a-repository/enabling-required-reviews-for-pull-requests
> In the Required approving reviews drop-down menu, select the number of approving reviews you'd like to require on the branch. 

also:

> Optionally, select Dismiss stale pull request approvals when new commits are pushed. This dismisses a pull request approval review when a code-modifying commit is pushed to the branch. 

Approval count is a good info, though!

**Comment by maximbaz** [comment]
Good point 🤔 I agree we probably shouldn't make any assumptions and should just return counts, maybe a count [for each of the possible approval states](https://docs.github.com/en/graphql/reference/enums#pullrequestreviewstate), since you'd possibly would query all of them anyway?
